### PR TITLE
MEN-2511: meta-mender updates for Yocto warrior

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc [hunk 1]
```diff
@@ -19,7 +19,6 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/patches:${THISDIR}/files:"
 SRC_URI_append_mender-uboot = " file://0001-Add-missing-header-which-fails-on-recent-GCC.patch"
 SRC_URI_append_mender-uboot = " file://0002-Generic-boot-code-for-Mender.patch"
 SRC_URI_append_mender-uboot = " file://0003-Integration-of-Mender-boot-code-into-U-Boot.patch"
-SRC_URI_append_mender-uboot = " file://0006-env-Kconfig-Add-descriptions-so-environment-options-.patch"
```
Comment (kacf) [inline meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc]: Should also remove the patch file.

File: meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-CONFIGS-rpi-enable-mender-requirements.patch [hunk 2]
```diff
@@ -11,73 +11,131 @@ Mender also requires that the environment is on MMC
...
+index 39da54c3e7..9a0fc1f0b5 100644
 --- a/configs/rpi_0_w_defconfig
 +++ b/configs/rpi_0_w_defconfig
-@@ -25,3 +25,4 @@ CONFIG_SYS_WHITE_ON_BLACK=y
+@@ -16,8 +16,6 @@ CONFIG_CMD_USB=y
+ CONFIG_CMD_FS_UUID=y
+ CONFIG_OF_EMBED=y
+ CONFIG_DEFAULT_DEVICE_TREE="bcm2835-rpi-zero-w"
+-CONFIG_ENV_FAT_INTERFACE="mmc"
+-CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
```
Comment (kacf) [inline meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-CONFIGS-rpi-enable-mender-requirements.patch]: Instinctively, this looks like a wrong change to me, but maybe it has been moved somewhere else? This should probably be tested on hardware to be sure, especially that `fw_setenv` works and is reflected at the boot prompt.

File: meta-mender-core/recipes-bsp/grub/patches/0001-workaround-wrong-detection-headeris-in-warrior.patch [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+@@ -74,9 +74,10 @@
+ /* Include the original <wchar.h> if it exists.
+    Some builds of uClibc lack it.  */
+ /* The include_next requires a split double-inclusion guard.  */
+-#if @HAVE_WCHAR_H@
++/* Assume the header file exists. For some strange reason,
++ * configure resolves the header to no exist when building
++ * on Yocto warrior branch */
+ # @INCLUDE_NEXT@ @NEXT_WCHAR_H@
+-#endif
```
Comment (kacf) [inline meta-mender-core/recipes-bsp/grub/patches/0001-workaround-wrong-detection-headeris-in-warrior.patch]: Is it difficult to fix `configure.ac` instead? I assume that such a patch is much more likely to be accepted upstream.

File: meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-CONFIGS-rpi-enable-mender-requirements.patch (hunk 2 above)
Comment (mirzak) [reply meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-CONFIGS-rpi-enable-mender-requirements.patch]: Above will try to load the U-boot environment from an file on the fat partition, which is not our intention.

This has always been there, but on previous U-boot versions the config `CONFIG_ENV_IS_IN_MMC=y` which we set seems to have taken precedence, but on this U-boot version it tried using FAT if this was not removed. There is a another change further down in this patch which also removes,

```
config ENV_IS_IN_FAT
...
default y if ARCH_BCM283X
```

Which is new for this U-boot version as well.

File: meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc]: Yes, I will

File: meta-mender-core/recipes-bsp/grub/patches/0001-workaround-wrong-detection-headeris-in-warrior.patch (hunk 3 above)
Comment (lluiscampos) [reply meta-mender-core/recipes-bsp/grub/patches/0001-workaround-wrong-detection-headeris-in-warrior.patch]: I tried hard! I really couldn't figure it out, specially the second one, the header `sysmacros.h`. For this first one, I located where the corresponding code in autotools live so I could hack it there, but finally just for being consistently hacky I decided to patch the source code.

ATM I am dealing with other issues (tests failing and so on) so I will keep this workaround for now and give it a second go at the end of my task.

File: meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-CONFIGS-rpi-enable-mender-requirements.patch (hunk 2 above)
Comment (kacf) [reply meta-mender-raspberrypi/recipes-bsp/u-boot/patches/0001-CONFIGS-rpi-enable-mender-requirements.patch]: You're right, easy to misread those defines. Disregard my comment then.

File: meta-mender-core/recipes-bsp/grub/patches/0001-workaround-wrong-detection-headeris-in-warrior.patch (hunk 3 above)
Comment (kacf) [reply meta-mender-core/recipes-bsp/grub/patches/0001-workaround-wrong-detection-headeris-in-warrior.patch]: Lol, "consistently hacky"! Sounds like it's more difficult than it's worth then.

**Comment by lluiscampos** [comment]
History of CI builds (for future reference)
- First full build [GitLab 82635274](https://gitlab.com/Northern.tech/Mender/mender-qa/pipelines/82635274) -> bitbake AUTOREV failure
- With updated warrior [GitLab 82713748](https://gitlab.com/Northern.tech/Mender/mender-qa/pipelines/82713748) -> same issue
- With latest yocto master [GitLab 83117641](https://gitlab.com/Northern.tech/Mender/mender-qa/pipelines/83117641) -> _in progress_

**Comment by lluiscampos** [comment]
@kacf This is now ready for review :)

The previous CI run was successful, and the latest push only added beatifications. I anyway triggered a new build just to make sure the bitbake bug is not affecting us.

The PR will go together with https://github.com/mendersoftware/mender-qa/pull/306

**Comment by lluiscampos** [comment]
Rebased on top of latest master (namely, getting 2.1 release) and running one last build: [GitLab 84017948](https://gitlab.com/Northern.tech/Mender/mender-qa/pipelines/84017948)

**Comment by kacf** [comment]
:champagne:

**Pull request merged**